vumi/transports/smpp/processors/sixdee.py

```diff
@@ -0,0 +1,216 @@
+# -*- test-case-name: vumi.transports.smpp.tests.test_sixdee -*-
+
+from vumi.config import ConfigInt
+from vumi.components.session import SessionManager
+from vumi.message import TransportUserMessage
+from vumi.transports.smpp.processors import default
+from vumi import log
+
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+
+def make_vumi_session_identifier(msisdn, sixdee_session_identifier):
+    return '%s+%s' % (msisdn, sixdee_session_identifier)
+
+
+class DeliverShortMessageProcessorConfig(
+        default.DeliverShortMessageProcessorConfig):
+
+    max_session_length = ConfigInt(
+        'Maximum length a USSD sessions data is to be kept for in seconds.',
+        default=60 * 3, static=True)
```

Comment:
We may want to revisit this default with this new gateway. We've been surprised before.